snowfall/training/mmi_graph.py

```diff
@@ -59,23 +60,21 @@ def __init__(self,
         oov:
           Out of vocabulary word.
         '''
-
-        L_inv = L_inv.to(device)
+        self.lexicon = lexicon
+        L_inv = self.lexicon.L_inv.to(device)
 
         if L_inv.properties & k2.fsa_properties.ARC_SORTED != 0:
             L_inv = k2.arc_sort(L_inv)
 
         assert L_inv.requires_grad is False
 
-        assert oov in words
+        assert oov in self.lexicon.words
 
         self.L_inv = L_inv
-        self.phones = phones
-        self.words = words
-        self.oov_id = self.words[oov]
+        self.oov_id = self.lexicon.words[oov]
         self.device = device
 
-        phone_symbols = get_phone_symbols(phones)
+        phone_symbols = get_phone_symbols(self.lexicon.phones)
```

Comment:
I think `self.lexicon.phone_symbols()` implements what `get_phone_symbols` is doing.

Reply:
Yeah I copied that function here - I’m trying to strike a balance between moving forward and not breaking every other training script that I didn’t update. I think after a bit more work we’ll be able to remove the left over code safely.